### ISSUE TITLE
TensorFlow - GPU Installation with RTX

### ISSUE THREAD

**Issue by omArvizu** [PyImageSearch/Deep-Learning-for-Computer-Vision-with-Python#1]
Well, Im trying to install tensorflow and keras from this guide, https://www.pyimagesearch.com/2019/01/30/ubuntu-18-04-install-tensorflow-and-keras-for-deep-learning/ because im following the ImageNet Bundle- Deep learning for computer vision with python course.. 

Im working from my personal computer, with ubuntu 18.04 in the graphic version, my GPU is a RTX3090 and my CPU is an AMD 3990x. so, at the beginning, when I just install ubuntu I installed the driver version 460.67.

+-----------------------------------------------------------------------------+
| NVIDIA-SMI 460.67 Driver Version: 460.67 CUDA Version: 11.2 |
|-------------------------------+----------------------+----------------------+
| GPU Name Persistence-M| Bus-Id Disp.A | Volatile Uncorr. ECC |
| Fan Temp Perf Pwr:Usage/Cap| Memory-Usage | GPU-Util Compute M. |
| | | MIG M. |
|===============================+======================+======================|
| 0 GeForce RTX 3090 Off | 00000000:01:00.0 On | N/A |
| 0% 27C P8 14W / 350W | 202MiB / 24265MiB | 0% Default |
| | | N/A |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes: |
| GPU GI CI PID Type Process name GPU Memory |
| ID ID Usage |
|=============================================================================|
| 0 N/A N/A 2352 G /usr/lib/xorg/Xorg 149MiB |
| 0 N/A N/A 2566 G /usr/bin/gnome-shell 41MiB |
| 0 N/A N/A 4046 G /usr/lib/firefox/firefox 4MiB |
| 0 N/A N/A 56057 G /usr/lib/firefox/firefox 4MiB |
+-----------------------------------------------------------------------------+

but you are working with the driver version 396, Im not sure if that might be the problem. 

when I installed tensorflow, the following error appear before the successfully installation message. 

ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
tensorflow-gpu 1.12.0 requires tensorboard<1.13.0,>=1.12.0, but you have tensorboard 2.4.1 which is incompatible.

So, I tried to uninstall tensorboard, with pip uninstall tensorboard 2.4.1 

" Successfully uninstalled tensorboard-2.4.1
WARNING: Skipping 2.4.1 as it is not installed."

then 

pip install tensorboard 1.13.0 but i got 

"ERROR: Could not find a version that satisfies the requirement 1.12.0
ERROR: No matching distribution found for 1.12.0" 

Now Im stocked here. =( 

any suggestion? kind regards!

I sent an email since past monday, and no one answer me =S I feel dissapointed because it suppose to have support from you and your team because I paid the course.. but I got promotional mails every single day

**Comment by sayakpaul** [comment]
Apologies for the delay. We are currently in between some migrations and hence the delay. 

Expect a reply from @abhishekthanki regarding this.

**Comment by abhishekthanki** [comment]
Hi @omArvizu,

Since you are using RTX 3090 GPU and it uses modern drivers (and would likely support only a modern version of CUDA). I think it's not possible to install TensorFlow 1.X GPU version using the official pre-compiled binaries as they are compatible with only specific versions of CUDA and that version would be quite old and mostly likely would not be compatible with your GPU. So, my suggestion would be to install and use TensorFlow 2.X. May I know which version of CUDA you have installed? Is it 11.2?

Best,
Abhishek

**Comment by omArvizu** [comment]
You are right, we have to use TF2, my only concern is that i will not be able to use this. 

> TFOD API is not compatible with TF 2.0 yet. For IB Chapters 14-17 you should use TF 1.14
Mask R-CNN chapters are not compatible with TF 2.0 yet.

I re install already evrything, ubuntu, drivers, cuda, etc.. and works fine with TF2

Regards

**Comment by sayakpaul** [comment]
> TFOD API is not compatible with TF 2.0 yet. 

TFOD API **_is_** compatible with TF2 now. Please check their [GitHub repository](https://github.com/tensorflow/models/tree/master/research/object_detection) to get the updates.
